### ISSUE TITLE
What you can do to improve Play!

### ISSUE THREAD

**Issue by Jordanesquivez** [jpd002/Play-#488]
Find the assets directory properly... this would result in various improvements, If qmake has been implemented u should Remove!. Remove unused sql database that are in the wrong place, cmake: if its implemented allow it to detect features at compile time, & also add ffmpeg submodule. For android you should change the preprocessor define to standard__ANDROID__, VU emulation should or can be improve.. and also add vertex caching (store the vertex data) and add frameskip (2fps).. I hope I am not asking too much as this work require alot of time.. I just wish that my suggestion can be implemented and wish it could work. Thanks for this wonderful emulator

**Comment by jpd002** [comment]
Yeah, improving VU emulation is high on the priority list. 
Vertex caching might help in some cases, but I need to get more data about that (GPU thread is rarely a bottleneck).
Frameskip is an interesting idea, but not sure it would help a lot (again, GPU thread is not a bottleneck most of the time).

**Comment by Jordanesquivez** [comment]
Oh I see.. so the gpu isn't a bottleneck at this time..... but a lot of glitches are in some games such as (need for speed underground and dbz budokai tenkaichi 3etc).. maybe its the texture replacement.. maybe?!...
oh try to add a texture replacer and for a little speed try to add an I or o (I/o) on thread.. as this contributed alot to the ppsspp emulator another thing what about trilinear texture.. this is the formula to a common artifact seen in mipmapped bilinearly filtered ...I've been doing alot of research.
  I learnt 4 months of emulation a textures nd coding

**Comment by Xstar97** [comment]
instead of using a sqlite database...have you thought about using Realm?
its like sqlite, but even better.
its free & the site will provide better info than me https://realm.io

**Comment by Fadi5555** [comment]
The big problem here is @jpd002 has a very limited time to work on this amazing project. So that's why this project is very slow to get even one single improve each month.
For long time I suggested to take this project somewhere or trying to attract more people who are always help each other on other projects like (retroarch, ppsspp, dolphin, pcsx2....etc) this is the best way to improve it. But now as you can see there is only one person who work on it is @jpd002 himself.
And as I said previously it has a very short time to work on it. If we enough care about this project we should find a solution for it. also you can take my suggestion above to deal with this issue.
Thanks.

**Comment by Jordanesquivez** [comment]
People should, and are helping.. but maybe a Little more  attention is needed from contributors to try to provide 2-3 improvements of speed and graphics in monthly builds... compatibility is very important but I would say some commercial games boot and goes ingame,people like to just enter the game and look at it the whole time but can't play it. Try to get a game to run at half speed or even at full speed if possible.. I will assume that if one game run at half speed other games of same type (as in graphical features and emulation elements) will too.. maybe an inspector or someone with more time should help out.. at least two days per week.. this would help out alot

**Comment by Fadi5555** [comment]
@Jordanesquivez you're absolutely right. But what's the method to attract people to work here? This is a question here?
Believe me if some people join here and start to work or at least @jpd002 himself work on it as you said 2 days weekly we could notice a lot of improvements even we can play some commercial games nearly to fullspeed. But nah. Unfortunately No one is care enough to work here.

**Comment by Jordanesquivez** [comment]
Yeah that's true.. I know a little about emulation and a little bit about ps2 hardware... plus gpu and everything about textures is my specialty I could help out... but In words I can suggest important key points on gpu to the emulator.. (people should show more interest)

**Comment by Xstar97** [comment]
I would love to believe me, but I do not have any experience with c/c++ &
iOS code....my knowledge is heavily based on Java which I am constantly
learning on a daily basis...if we can come up with a more universal system
to expand the emulator's capability such as:

- expanding compatibility across all devices & games.

- adding crash & log logic for users to send to the developer(s)
IE when a game crashes/doesn't work.
- diagnosing the device for any compatibility issues...IE ram, graphic,
storage, etc.

This will make easier for all the developers instead of playing the
"waiting & guessing" game & hoping it would work.

I written an app before without crash/log logic...so it was hard to
diagnose any issues without it...so any app/library I make I always add
crash logic to it...but I didn't see anything when I skimmed the code or
the app itself to send the crash data back to the developer.

**Comment by Jordanesquivez** [comment]
@Xstar97 I get your point.. like a std logging?? 
that might help in some consequences... but as I said if games can run at half speed it would really really be very good. oh and expanding game across all devices.. could or maybe cause slowdowns.. as the os might not be the problem but the gpu,ram,processor and storage capacity might be a problem ... devices with android os 4.0.3 to 7.0 may have high gpu,ram and fast processor but lower may be just too below the entailments..

**Comment by SolarstrikeX2** [comment]
I will do my best to report any issues and status updates on various games, despite of @jpd002's time constraints.

**Comment by minuteworld** [comment]
you people are also forgetting that you can just shill this github to emulation forums, imageboards, ask r/emulation if anyone wants to help (I doubt it) etc. I am really impressed by this emulator but it needs more people. it's the complete opposite of xqemu where on that, they've had 300+ contributors over the years and it's still unusable whereas this has one person on it, has potential to blast pcsx2 OUT OF THE WATER but noone wants to work on it but the creator whom I respect.

seriously, I respect jpd002 for his work. the code's good, you don't need a ps2 bios, etc. it's great.

**Comment by jpd002** [comment]
Sorry guys, I've been quite busy these past weeks. I've finally merged the "better_efficiency" branch which should help with performance on mobile. I'm still focusing on improving compatibility. I've also got plans to address the game info/cover database problem soon.

**Comment by Jordanesquivez** [comment]
@jpd002 thanks for your response... the branch"better efficiency" does it boost fps?? and by how much??

**Comment by Fadi5555** [comment]
@Jordanesquivez yeah me too I just want to ask this question as well.
@jpd002 we really hope to fix boxart on iOS if you have time because until now it's not showing covers of games. Also please add select and start to the ps3/4 controller. all buttons are mapping good except select and start.

Finally I'm really hope to enhancing fps of games as well as graphics issues on some games like (ground hunting, Siren, crash bandicoot the warth of cortex, and some more).
Good luck.

**Comment by Jordanesquivez** [comment]
@jpd002 when will new build be available

**Comment by atomic83GitHub** [comment]
There is now a discord server for talking about this. @jpd002, this can be close right?
